app/src/main/java/com/example/a310_rondayview/data/event/EventsFirestoreManager.java

```diff
@@ -0,0 +1,62 @@
+package com.example.a310_rondayview.data.event;
+
+import static com.example.a310_rondayview.data.event.EventsFirestoreContract.EVENT_COLLECTION_NAME;
+
+import com.example.a310_rondayview.Event;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.EventListener;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+
+public class EventsFirestoreManager {
+
+    private static EventsFirestoreManager eventsFirestoreManager;
```

Comment:
not sure too much about this, but apparently making this **volatile** is useful for thread-safety in singleton pattern
It ensures that changes made by one thread are visible to all other threads.
   
so basically this, private static volatile EventsFirestoreManager eventsFirestoreManager; but i think the current implementation also works fine, just something interesting i thought to share!

Reply:
😮 given the current implementation isn't multi-threaded I will keep it as this for now though :-)